homeassistant/components/axis/__init__.py

```diff
@@ -31,7 +31,9 @@ async def async_setup_entry(hass, config_entry):
 
     await device.async_update_device_registry()
 
-    hass.bus.async_listen_once(EVENT_HOMEASSISTANT_STOP, device.shutdown)
+    device.listeners.append(
+        hass.bus.async_listen_once(EVENT_HOMEASSISTANT_STOP, device.shutdown)
+    )
```

Comment:
You could consider unsubscribing in `async_unload_entry` instead, it makes it easier to follow the flow.

Reply:
Thanks! I will consider how to do this